### PR TITLE
Read-only is not working in file-attachment

### PR DESCRIPTION
**Pull request by pavi41**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Review by rismehta — CHANGES_REQUESTED** [review]
please add test cases for all the changes

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v2/fileinput/clientlibs/site/js/fileinputview.js [hunk 1]
```diff
@@ -86,6 +86,19 @@
             return this.element.querySelector(FileInputV2.selectors.attachButtonLabel);
         }
 
+        updateEnabled(enabled, state) {
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v2/fileinput/clientlibs/site/js/fileinputview.js:89]:
```
updateEnabled(enabled, state) {

    // Update data attribute for enabled state
    this.element.setAttribute(FormView.Constants.DATA_ATTRIBUTE_ENABLED, enabled);
    
    // Determine if the widget should be disabled
    const isDisabled = !enabled || state.readOnly;

    // Update the 'disabled' and ARIA 'disabled' attributes based on the enabled state
    if (isDisabled) {
        this.widget.setAttribute("disabled", "disabled");
        this.widget.setAttribute(FormView.Constants.ARIA_DISABLED, "true");
    } else {
        this.widget.removeAttribute("disabled");
        this.widget.removeAttribute(FormView.Constants.ARIA_DISABLED);
    }
}

```

Always try to create a constant inside the if condition to make the code more readable

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/js/dropdownview.js [hunk 2]
```diff
@@ -84,7 +84,6 @@
         }
 
         updateEnabled(enabled, state) {
-            this.toggle(enabled, FormView.Constants.ARIA_DISABLED, true);
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/js/dropdownview.js:87]: Why is this code removed ? Isn't this backward incompatible

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/js/dropdownview.js (hunk 2 above)
Comment (pavi41) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/js/dropdownview.js:87]: This line was adding aria-disabled attribute to the parent div which is of no use, so I has removed this line.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v2/fileinput/clientlibs/site/js/fileinputview.js (hunk 1 above)
Comment (pavi41) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v2/fileinput/clientlibs/site/js/fileinputview.js:89]: sure, implemented the same

File: ui.tests/test-module/specs/fileinput/fileinput.runtime.spec.js [hunk 3]
```diff
@@ -80,7 +80,7 @@ describe("Form with File Input - Basic Tests", () => {
     const checkHTML = (id, state) => {
         const visible = state.visible;
         const passVisibleCheck = `${visible === true ? "" : "not."}be.visible`;
-        const passDisabledAttributeCheck = `${state.enabled === false ? "" : "not."}have.attr`;
+        const passDisabledAttributeCheck = `${state.enabled === false || state.readOnly === true ? "" : "not."}have.attr`;
```
Comment (rismehta) [inline ui.tests/test-module/specs/fileinput/fileinput.runtime.spec.js:83]: As discussed in the scrum, can you please add the same check in all the test specs for all the components ?

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.tests/test-module/specs/fileinput/fileinput.runtime.spec.js (hunk 3 above)
Comment (pavi41) [reply ui.tests/test-module/specs/fileinput/fileinput.runtime.spec.js:83]: Sure Rishi, Will be adding the test cases in the next pr which I am working on for tnc component

**Pull request merged**